### PR TITLE
11377 - WS Devops deployment of Helm charts with Provenance and Container instance via Shell stage

### PR DESCRIPTION
**Pull request by RahulMR42**
# Add here your WMS ID

# General requirements
1. Information in the workshop is adequate and updated
2. Code is correct and working
3. Links are correct
4. Make sure you enter the help email link in your manifest.json
5. Please make sure WMS URLs are updated as needed after your PR is approved

# Checklist - Refer to the QA document for the complete list 
Please confirm that the following is completed before submitting your PR

- [x] All filenames are lower case (including folders, images, files, etc.)
- [x] Filenames are descriptive
- [x] Your workshop folder structure should be similar to the one used in the sample workshop (https://github.com/oracle-livelabs/common/tree/main/sample-livelabs-templates/sample-workshop)
- [x] Are you using multiple versions (desktop/, sandbox/, tenancy/)? Make sure that each of them contains a manifest.json and an index.html
- [x] Image references in markdown contain an alternate text

### CODE REVIEW

File: oci-devops-cicd-helm-ci/intro/intro.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+- File-based trigger of OCI Build pipeline. : In Oracle Cloud Infrastructure (OCI) DevOps, a build run can be automatically triggered when you commit your changes to a code repository. Using this we can set the exclusion and inclusion of a set of files /folders against which the build pipelines execute during a code change.
+- Managed build stage using custom build runner. : Custom build runner allows to set desired OCPU and Memory for the build runner to support necessary resources for build instructions to run.
+- Using terraform with Remote state backend from OCI DevOps build pipeline. : The S3-compatible OCI Object storage bucket will be used as a remote backend to store the terraform states. The terraform will be executed from the OCI Build pipeline, which is enabled with resource principal so that the client can control the resource access based on policies.
+
+### Prerequisites
+
+1. An Oracle Free Tier(Trial), Paid or LiveLabs Cloud Account
```
Comment (anooshapilli) [inline oci-devops-cicd-helm-ci/intro/intro.md:28]: @RahulMR42 - Make the Prerequisites points bullets

File: oci-devops-cicd-helm-ci/intro/intro.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
+
+# Introduction
+
+## About this workshop.
+
+You will learn to use OCI DevOps helm deployment with provenance and integrity checks as well as deployment of a sample application onto Oracle container instances.
+
+Estimated total time: 90 minutes
+
+### Objectives
```
Comment (anooshapilli) [inline oci-devops-cicd-helm-ci/intro/intro.md:10]: @RahulMR42 - Move the objectives section before the Prerequisites section

File: oci-devops-cicd-helm-ci/helm-deployment/helm-deployment.md [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+
+## Learn More
+
+* [OCI Devops documentation](https://docs.oracle.com/en-us/iaas/Content/devops/using/home.htm)
+
+
+## Acknowledgements
+
+* **Author** - Rahul M R
+* **Contributors** -
```
Comment (anooshapilli) [inline oci-devops-cicd-helm-ci/helm-deployment/helm-deployment.md:222]: @RahulMR42 - Add contributor details. If there are no contributors, you can remove that line.

File: oci-devops-cicd-helm-ci/helm-deployment/helm-deployment.md [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+## Task 1: GPG Key Setup (For Helm Signing and Verification).
+
+1. We are using GnuPG (gpg) to set up the key for signing and verifying the helm chart. Read more about gpg [here](https://gnupg.org/).
+
+1. We are going to install gpg and the instruction here is related to a Linux machine. You may change the procedure accordingly - https://www.gnupg.org/howtos/card-howto/en/ch02.html.
+
+
+1. Follow the below instructions and download the stable version of gpg onto your workstation.
+
+```java
```
Comment (anooshapilli) [inline oci-devops-cicd-helm-ci/helm-deployment/helm-deployment.md:27]: @RahulMR42 Add copy button code blocks to the code and indent them with 4 spaces from begin of line. For more details, you can refer the guide - https://oracle-livelabs.github.io/common/sample-livelabs-templates/create-labs/labs/workshops/livelabs/?lab=4-labs-markdown-develop-content#Task12:CodeSnippets

**Review by anooshapilli — COMMENTED** [review]
@RahulMR42 - There are changes made to different folders other than the 11377 workshop folder. Could you share the WMS ID for that workshop?
image

Please make the requested changes to the workshop

**Comment by RahulMR42** [comment]
> 

The

File: oci-devops-cicd-helm-ci/workshops/livelabs/manifest.json [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+      "title": "Lab5: Cleanup of lab resources",
+      "description" : "Clean all the lab infra components",
+      "filename":"../../cleanup/cleanup.md"
+
+    },
+
+    {
+      "title": "Need Help?",
+      "description": "Solutions to Common Problems and Directions for Receiving Live Help",
+      "filename": "https://oracle-livelabs.github.io/common/labs/need-help/need-help-freetier.md"
```
[the remark this answers is not in the document]
Comment (RahulMR42) [reply oci-devops-cicd-helm-ci/workshops/livelabs/manifest.json:45]: Done.

File: oci-devops-cicd-helm-ci/workshops/desktop/manifest.json [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+      "title": "Lab5: Cleanup of lab resources",
+      "description" : "Clean all the lab infra components",
+      "filename":"../../cleanup/cleanup.md"
+
+    },
+
+    {
+      "title": "Need Help?",
+      "description": "Solutions to Common Problems and Directions for Receiving Live Help",
+      "filename": "https://oracle-livelabs.github.io/common/labs/need-help/need-help-freetier.md"
```
[the remark this answers is not in the document]
Comment (RahulMR42) [reply oci-devops-cicd-helm-ci/workshops/desktop/manifest.json:45]: Done

File: oci-devops-cicd-helm-ci/shellstage-deployment/shellstage-deployment.md [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+In this lab, as a developer or SRE,
+
+* Run OCI Build and deploy pipeline.
+* Validate the container instance deployment
+
+## Task 1: Run OCI Build and deploy pipeline.
+
+1. Using OCI Console > `Developer Services` > `Devops` >` Projects`.Click on the project name.
+
+![oci-devops-projects.png](images/oci-devops-projects.png)
```
[the remark this answers is not in the document]
Comment (RahulMR42) [reply oci-devops-cicd-helm-ci/shellstage-deployment/shellstage-deployment.md:21]: Done

File: oci-devops-cicd-helm-ci/shellstage-deployment/shellstage-deployment.md [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+
+## Learn More
+
+* [OCI Devops documentation](https://docs.oracle.com/en-us/iaas/Content/devops/using/home.htm)
+
+
+## Acknowledgements
+
+* **Author** - Rahul M R
+* **Contributors** -
```
[the remark this answers is not in the document]
Comment (RahulMR42) [reply oci-devops-cicd-helm-ci/shellstage-deployment/shellstage-deployment.md:79]: None for now

File: oci-devops-cicd-helm-ci/intro/intro.md (hunk 1 above)
Comment (RahulMR42) [reply oci-devops-cicd-helm-ci/intro/intro.md:28]: Done

File: oci-devops-cicd-helm-ci/helm-deployment/helm-deployment.md (hunk 3 above)
Comment (RahulMR42) [reply oci-devops-cicd-helm-ci/helm-deployment/helm-deployment.md:222]: Done

File: oci-devops-cicd-helm-ci/helm-deployment/helm-deployment.md (hunk 4 above)
Comment (RahulMR42) [reply oci-devops-cicd-helm-ci/helm-deployment/helm-deployment.md:27]: Done

**Comment by RahulMR42** [comment]
@anooshapilli 
Thank you for the reviews, the fixes are applied.
There are minor edits towards the ID 3324 and new addition for ID 11377

**Pull request merged**